src/client.cpp

```diff
@@ -194,22 +194,25 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
     return value.HasMember(name)  &&  value[name].IsInt();
 }
 
+#define CREATE_SESSION() \
+    CurlSessionPtr sessionPtr = createSession();\
+    if (!sessionPtr)\
+        return makeError();\
+    CurlSession& session = *sessionPtr;
```

Comment:
I don't like the idea that you have to know what `CREATE_SESSION` is doing in order to use it in the right way.
I would leave it as it was. There is no much code duplication here and that code was clear.
BTW you still have direct `createSession()` calls in the file.